src/ChainRec.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var assert = require('./internal/assert');
+var chain = require('./internal/chain');
+var chainRec = require('./internal/chainRec');
+var map = require('./internal/map');
+
+
+module.exports = function(equals, CR) {
+  return {
+
+    equivalence: assert.forall1(function(i) {
+      var p = '???';
+      var n = '???';
+      var d = '???';
+      var lhs = chainRec(CR)(function(next, done, v) { return p(v) ? map(done)(d(v)) : map(next)(n(v)); })(i);
+      var rhs = (function step(v) { return p(v) ? d(v) : chain(step)(n(v)); }(i));
+      return equals(lhs, rhs);
+    })
```

Comment:
As with all things ChainRec I'll defer to you on this, @safareli. Should `p`, `n`, and `d` be provided by the caller?